### PR TITLE
Add comprehensive test coverage for messaging, components, and utilities

### PR DESCRIPTION
**Pull request by EspritVorace**
## Summary
This PR significantly expands test coverage across the codebase by adding new test files and enhancing existing ones. The changes focus on testing background messaging logic, Storybook component stories, and utility functions with improved error handling scenarios.

### New Test Files
- **`tests/background/messaging.test.ts`**: Comprehensive test suite for background messaging functions including:
  - `handleMiddleClickMessage`: Tests for URL registration and error handling
  - `handleSessionRestoreSkipDedupMessage`: Tests for deduplication marking
  - `cleanupMiddleClickedTabsForTab`: Tests for tab cleanup logic
  - `promptForGroupName`: Tests for group name prompting with validation
  - `findMiddleClickOpener`: Tests for middle-click opener detection with fallback logic

- **`tests/components/FileDropZone.stories.test.tsx`**: Portable Storybook story tests for the FileDropZone component covering empty state, file display, and button interactions

- **`tests/components/DomainRuleCard.stories.test.tsx`**: Portable Storybook story tests for DomainRuleCard covering default, disabled, drag-disabled, and search highlight states

- **`src/components/UI/ImportExportWizards/Source/FileDropZone.stories.tsx`**: New Storybook stories for FileDropZone component with empty, with-file, and drag-over variants

### Enhanced Test Coverage
- **`tests/components/DomainRulesPage.stories.test.tsx`**: Added three new test cases:
  - Empty state display when search returns no results
  - Single rule deletion dialog opening
  - Bulk rule deletion dialog opening

- **`tests/tabCapture.test.ts`**: Added two new test cases:
  - Color normalization validation
  - Graceful handling when tab group retrieval fails

- **`tests/utils/statisticsUtils.test.ts`**: Added two new test cases:
  - Error handling when storage write fails in `resetStatisticsData`
  - Callback invocation with merged data in `watchStatisticsData`

- **`tests/utils/settingsUtils.test.ts`**: Added two new test cases:
  - Error handling when storage write fails in `setSettings`
  - Error handling when storage write fails in `updateSettings`

## Notable Implementation Details
- All new tests use proper mocking with `vi.mock()` for external dependencies
- Tests follow French naming conventions for test descriptions (matching existing codebase style)
- Comprehensive error scenario coverage including storage failures and missing data
- Portable Storybook stories enable testing component behavior in isolation
- Tests validate both happy paths and edge cases (missing IDs, empty values, rejected promises)

https://claude.ai/code/session_01587WgTHBzwxPCuaEN7LEeG

### CODE REVIEW

File: tests/pages/popup.test.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      expect(browser.tabs.create).toHaveBeenCalledWith(
+        expect.objectContaining({ url: expect.stringContaining('#rules') }),
+      );
+    });
+    expect(window.close).toHaveBeenCalled();
+  });
+
+  it("met à jour l'onglet existant si un onglet options est déjà ouvert", async () => {
+    mockedTabsQuery.mockResolvedValue([{ id: 42, windowId: 1 }]);
+    render(<PopupApp />);
```
Comment (EspritVorace) [inline tests/pages/popup.test.tsx:147]: why asserting only 1 thing at a time ? It implies too many renders and sliw tests
